module/VuFind/src/VuFind/Db/Service/SearchService.php

```
@@ -118,7 +119,7 @@ public function destroySession(string $sessionId, UserEntityInterface|int|null $
      */
     public function getSearchById(int $id): ?SearchEntityInterface
     {
-        return $this->getDbTable('search')->select(['id' => $id])->current();
+        return $this->entityManager->find(SearchEntityInterface::class, $id);
```

@demiankatz, this line raises a runtime exception in my local test installation:

`Ausnahme: Doctrine\Persistence\Mapping\MappingException
Class 'VuFind\Db\Entity\SearchEntityInterface' does not exist`

Should it be `Search::class` or `\VuFind\Db\Entity\Search::class` here (and in the rest of the file)?

---

The SearchEntityInterface is defined in https://github.com/vufind-org/vufind/blob/dev/module/VuFind/src/VuFind/Db/Entity/SearchEntityInterface.php, and [SearchEntityInterface is set up as an alias in the PluginManager](https://github.com/vufind-org/vufind/blob/dev/module/VuFind/src/VuFind/Db/Entity/PluginManager.php#L66), so I'm not sure why you are encountering a problem. I would expect the code to work as written (and indeed, if it didn't, I would expect some of our integration tests -- particularly [SavedSearchesTest](https://github.com/vufind-org/vufind/blob/dev/module/VuFind/tests/integration-tests/src/VuFindTest/Mink/SavedSearchesTest.php) -- to fail).

I'll try to find time to double-check this more carefully in my own test environment when I'm back in the office tomorrow. In the meantime, is there anything unusual about your environment? Can you specify what steps you're taking to raise the exception? Are you able to run integration tests?

---

I get more similar exceptions ("Class 'VuFind\Db\Entity\ResourceEntityInterface' does not exist" and others).

---

Does the whole system fail to work, or are you just seeing these types of errors occasionally? What PHP version are you using? Is there any possibility that your Composer dependencies are out of date or that there's something odd in your cache?

---

I don't think that macOS is a very common environment for VuFind tests, so yes, that's unusual. The integration tests (`vendor/bin/phing qa-php`) show lots of messages like `Unable to connect to localhost:80 (Connection refused)`, so it is also unusual that the Homebrew Apache listens on port 8080.

---

I believe that @EreMaijala has had success getting things working in the environment, so he might have some advice. Have you seen the notes on the [testing wiki page](https://vufind.org/wiki/development:testing:unit_tests)?

---

That exception happens at runtime if you catch all exceptions in a debugger, but does not affect functionality.

---

I catch "everything" (which includes all exceptions) in my debugger because it typically indicates errors or at least weaknesses in the code. Here an AI chatbot also says that looking for a class with the name of the interface is wrong, and the name of the class which is derived from the interface should be passed instead.

---

@stweil, it is conventional when using Laminas service managers to name services for interfaces, to make it easier to install local custom versions without having to alias concrete classes in potentially confusing ways. That is the pattern that has been followed here, and it is set up in our custom [EntityManagerFactory](https://github.com/vufind-org/vufind/blob/dev/module/VuFind/src/VuFind/Db/EntityManagerFactory.php). This may be a conflict between Laminas practices and Doctrine practices, but it was an intentional choice based on Laminas practice.

I don't really have strong feelings about this one way or the other; I think the likelihood of locally overriding entities is relatively small. However, it is a fairly large change to make this close to a major release, so I would be hesitant to change it further unless there's a very strong reason to do so. Maybe the thing worth investigating further is why exceptions are being thrown and caught when aliases are properly set up -- but I imagine that would require a deep dive into the Doctrine code.